Feat/598 connect qlik sense desktop

r-hannuschka opened the pull request.

konne reviewed (changes requested):
@r-hannuschka please let us discuss to avoid the isQlikSenseDesktop boolean
It should be possible with all existing configs just to set everything right from the settings view.

r-hannuschka commented:
I dont see many options here

qlik sense desktop:

Server is running at localhost[:PORT]/hub
Websocket is running at localhost[:4848]

qlik sense server / docker

Server is running at localhost[:port]
Websocket is running at localhost[:port]

In Docker / Qlik Sense Server, the Websocket and Server running on same url not so in Qlik Sense Desktop thats makes it a bit complicated since we need to add the hub.

Current flow is, check the server exists and get certificates for it (if https), we need an additional step for QlikSenseDesktop then which checks which url will answer.

1. running against localhost:4848 (QlikSenseServer/ Docker)
2. if this fails try localhost:4848/hub (QlikSenseDesktop)

r-hannuschka commented:
Removed all settings for isQlikSenseDesktop now, since we dont save it in the package.json we could not edit a connection correctly anymore since the information gets lost.

Updated this to this, that we now run 2 loops:

for example this is qlikSenseDesktop connection:

host: localhost
port: 4848

first attempt: localhost:4848 if this fails it runs a second attempt
second attempt: localhost:4848/hub if this fails server just not exists
